zoltar-core/src/main/java/com/spotify/zoltar/fs/FileSystemExtras.java

```diff
@@ -98,19 +98,21 @@ public static URI downloadIfNonLocal(final URI path) throws IOException {
   }
 
   static Path copyDir(final Path src, final Path dest, final boolean overwrite) throws IOException {
-    final List<URI> uris =
+    final List<Path> paths =
         Files.walk(src)
             .filter(path -> !path.equals(src))
-            .map(Path::toUri)
             .collect(Collectors.toList());
 
-    for (final URI uri : uris) {
-      final String relative =
-          uri.toString().substring(src.toUri().toString().length(), uri.toString().length());
-      final Path fullDst = Paths.get(dest.toUri().resolve(relative));
+    for (final Path path : paths) {
+      final Path relative = src.relativize(path);
+      // The 'resolve' method can be passed a String or a Path - we must pass String. If copying
+      // from a jar file, the relative path will be ZipPath, while our destination directory will
+      // be UnixPath. This difference will cause resolve to throw ProviderMismatchException, so
+      // we must first convert relative to String.
+      final Path fullDst = dest.resolve(relative.toString());
       final CopyOption[] flags =
-          overwrite ? new CopyOption[] {StandardCopyOption.REPLACE_EXISTING} : new CopyOption[] {};
```

Comment:
@rculbertson can you run spotless? `mvn spotless:apply` ... not sure why build didn't fail on travis ... I need to have a look at it

Reply:
Fixed.